redundant-bmc/src/manager.cpp

```diff
@@ -277,9 +277,31 @@ void Manager::disableRedPropChanged(bool disable)
 }
 
 // NOLINTNEXTLINE
-sdbusplus::async::task<> Manager::method_call(
-    start_failover_t /* unused */, const FailoverOptions& /* options */)
+sdbusplus::async::task<> Manager::method_call(start_failover_t /* unused */,
+                                              const FailoverOptions& options)
 {
+    if (!handler)
+    {
+        lg2::error("Failover not allowed because it is too early");
+        throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
+    }
+
+    if (redundancyInterface.failover_imminent() || failover_in_progress())
+    {
+        lg2::error(
+            "Failover not allowed because a failover is already imminent or in progress ");
+        throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
```

Comment:
I think we should define a separate error for this or we can revisit it later if the DBus caller needs to handle the error differently.